cancer_ai/validator/competition_manager.py

```diff
@@ -159,56 +167,60 @@ async def sync_chain_miners(self):
-            )
-            if not hotkey_metadata:
-                bt.logging.warning(
-                    f"Cannot get miner model for hotkey {hotkey} from the chain, skipping"
-                )
+        for hotkey in self.chain_miner_models.hotkeys:
+            if not self.chain_miner_models.hotkeys[hotkey]:
                 continue
-            try:
-                miner_model = await self.get_miner_model(hotkey)
-                self.chain_miner_models[hotkey] = hotkey_metadata
-                self.model_manager.hotkey_store[hotkey] = miner_model
-            except ValueError:
-                bt.logging.error(
-                    f"Miner {hotkey} with data  {hotkey_metadata.to_compressed_str()} does not belong to this competition, skipping"
+            model_info = await self.chain_miner_to_model_info(
+                self.chain_miner_models.hotkeys[hotkey]
+            )
+            if model_info:
+                self.model_manager.hotkey_store[hotkey] = (
+                    await self.chain_miner_to_model_info(
+                        self.chain_miner_models.hotkeys[hotkey]
+                    )
                 )
+        hotkeys_with_models = list(
+            filter(
+                lambda k: self.model_manager.hotkey_store[k],
+                self.model_manager.hotkey_store,
+            )
+        )
         bt.logging.info(
-            f"Amount of chain miners with models: {len(self.chain_miner_models)}"
+            f"Amount of hotkeys with valid models: {len(hotkeys_with_models)}"
         )
 
-    async def evaluate(self) -> str:
+    async def evaluate(self) -> str | None:
         """Returns hotkey and competition id of winning model miner"""
+        bt.logging.info(f"Start of evaluation of {self.competition_id}")
+        if self.test_mode:
+            await self.sync_chain_miners_test()
+        else:
+            await self.sync_chain_miners()
+        if len(self.model_manager.hotkey_store) == 0:
+            bt.logging.error("No models to evaluate")
+            return None
         await self.dataset_manager.prepare_dataset()
         X_test, y_test = await self.dataset_manager.get_data()
 
         competition_handler = COMPETITION_HANDLER_MAPPING[self.competition_id](
             X_test=X_test, y_test=y_test
         )
-        # TODO get hotkeys
-        if self.test_mode:
-            await self.sync_chain_miners_test()
-        else:
-            await self.sync_chain_miners()
 
         y_test = competition_handler.prepare_y_pred(y_test)
-        # bt.logging.info("Ground truth: ", y_test)
         for hotkey in self.model_manager.hotkey_store:
-            bt.logging.info("Evaluating hotkey: ", hotkey)
-            await self.model_manager.download_miner_model(hotkey)
-
+            bt.logging.info(f"Evaluating hotkey: {hotkey}")
+            was_downloaded = await self.model_manager.download_miner_model(hotkey)
+            if not was_downloaded:
```

Comment:
Does it makes sense to have nested try except statement pattern here? The download_miner_model is using try/except. We could then take over this potential raise and show an error here instead of this conditional logic?